src/wire_msg.rs

```diff
@@ -22,25 +20,27 @@ pub enum WireMsg {
     UserMsg(bytes::Bytes),
 }
 
-impl Into<bytes::Bytes> for WireMsg {
-    fn into(self) -> bytes::Bytes {
-        if let WireMsg::UserMsg(ref m) = self {
-            if m.len() > MAX_MESSAGE_SIZE_FOR_SERIALISATION {
-                return m.clone();
-            }
-        }
+const USER_MESSAGE: u8 = 0;
+const WIRE_MESSAGE: u8 = 1;
```

Comment:
Even a user msg is a wire msg, so this naming convention is a bit strange.
```suggestion
const USER_MSG_FLAG: u8 = 0;
```
Then for non-user-msg, just do `!USER_MSG_FLAG`. So no need of two variants.